rust/api.md

```diff
@@ -0,0 +1,532 @@
+# Definition for the new Scanner HTTP API via JSON
+
+
+## Requests
+
+### start scan
+
+The start scan command is used to start a scan. It is important to use a unique scan ID. The response of this request just returns a status code wether it worked or not, without any data.
+
+- Path: `/scan/start`
+- Request:
+<pre>
+  <code>
+    {
+      <a href="#scan_id">"scan_id"</a>: string,
+      <a href="#targets">"targets"</a>: [{
+        <a href="#target">"target"</a>: [string],
+        <a href="#ports">"ports"</a>: [string],
+        <a href="#credentials">"credentials"</a>: [{
+          <a href="#service">"service"</a>: string,
+          <a href="#port">"port"</a>: int,
+          <a href="#up">"up"</a>: {
+            <a href="#username">"username"</a>: string,
+            <a href="#password">"password"</a>: string
+          },
+          <a href="#usk">"usk"</a>: {
+            <a href="#username">"username"</a>: string,
+            <a href="#password">"password"</a>: string,
+            <a href="#private">"private"</a>: string
+          },
+          <a href="#snmp">"snmp"</a>: {
+            <a href="#username">"username"</a>: string,
+            <a href="#password">"password"</a>: string,
+            <a href="#community">"community"</a>: string,
+            <a href="#auth_algorithm">"auth_algorithm"</a>: string,
+            <a href="#privacy_password">"privacy_password"</a>: string,
+            <a href="#privacy_algorithm">"privacy_algorithm"</a>: string
+          }
+        }]
+      }],
+      <a href="#excluded">"excluded"</a>: [string],
+      <a href="#scanner_parameters">"scanner_parameters"</a>: {key:value},
+      <a href="#vts">"vts"</a>: {
+        <a href="#vt_single">"vt_single"</a>: [{
+          <a href="#oid">"oid"</a>: string,
+          <a href="#vt_parameters">"vt_parameters"</a>: [{
+            <a href="#parameter">"parameter"</a>: string,
+            <a href="#value">"value"</a>: any
+          }]
+        }],
+        <a href="#vt_group">"vt_group"</a>: [string]
+      }
+    }
+  </code>
+</pre>
+- Status codes: 
+  - 200 Ok
+  - 400 Bad Request
+- Response: None
```

Comment:
Since it is important to use a unique id I think it would be better to generate and return the ID in the backend rather than let the client choose one.

Reply:
For debug purposes I'd like to be able to set the scan ID myself, so it is possible to have scripts that get scan information and delete scans without having to adjust them for each scan.
But yes we could make the scan ID optional and response with one if none was given

Reply:
But you can easily write script by parsing the returned json?

Reply:
Just for the record. In the current GVM stack, the scan id is generated and sent by client and corresponds to the report ID. The same Report ID is sent in case of a task resume.

Reply:
Yeah, I am aware of that. This approach however seems to be very error prone and the danger of loosing data is high. I think it is better to be able to implement the start/stop/resume mechanism directly into the scanner and let the scanner directly handle all data that is necessary. I think this would overall simplify this functionality and the client does not have to bother about saving data that is only necessary for this mechanic. Additionally it would make the scanner more reliable when saving such data to a storage, so it is able to recover from crashes without having the client to react with some actions.